src/server/scripts/Events/midsummer.cpp

```diff
@@ -49,6 +68,319 @@ class go_midsummer_bonfire : public GameObjectScript
+    { { 571, 66,   TEAM_ALLIANCE }, &BonfireStampedOutState[64] },
+    { { 571, 66,   TEAM_HORDE    }, &BonfireStampedOutState[65] },
+    { { 571, 67,   TEAM_ALLIANCE }, &BonfireStampedOutState[66] },
+    { { 571, 67,   TEAM_HORDE    }, &BonfireStampedOutState[67] },
+    { { 571, 394,  TEAM_ALLIANCE }, &BonfireStampedOutState[68] },
+    { { 571, 394,  TEAM_HORDE    }, &BonfireStampedOutState[69] },
+    { { 571, 495,  TEAM_ALLIANCE }, &BonfireStampedOutState[70] },
+    { { 571, 495,  TEAM_HORDE    }, &BonfireStampedOutState[71] },
+    { { 571, 2817, TEAM_ALLIANCE }, &BonfireStampedOutState[72] },
+    { { 571, 2817, TEAM_HORDE    }, &BonfireStampedOutState[73] },
+    { { 571, 3537, TEAM_ALLIANCE }, &BonfireStampedOutState[74] },
+    { { 571, 3537, TEAM_HORDE    }, &BonfireStampedOutState[75] },
+    { { 571, 3711, TEAM_ALLIANCE }, &BonfireStampedOutState[76] },
+    { { 571, 3711, TEAM_HORDE    }, &BonfireStampedOutState[77] },
+};
+
+const uint32 GoBonfireAlliance[COUNT_GO_BONFIRE_ALLIANCE] = { 187564, 187914, 187916, 187917, 187919, 187920, 187921, 187922, 187923, 187924, 187925, 187926, 187927, 187928, 187929, 187930, 187931, 187932, 187933, 187934, 187935, 187936, 187937, 187938, 187939, 187940, 187941, 187942, 187943, 187944, 187945, 187946, 194032, 194035, 194036, 194038, 194040, 194044, 194045, 194049 };
+const uint32 GoBonfireHorde[COUNT_GO_BONFIRE_HORDE] = { 187559, 187947, 187948, 187949, 187950, 187951, 187952, 187953, 187954, 187955, 187956, 187957, 187958, 187959, 187960, 187961, 187962, 187963, 187964, 187965, 187966, 187967, 187968, 187969, 187970, 187971, 187972, 187973, 187974, 187975, 194033, 194034, 194037, 194039, 194042, 194043, 194046, 194048 };
+const uint32 GoBonfireCity[COUNT_GO_BONFIRE_CITY] = { 181332, 181333, 181334, 181335, 181336, 181337, 188128, 188129, 188352 };
+
+class MidsummerPlayerScript : public PlayerScript
+{
+public:
+    MidsummerPlayerScript() : PlayerScript("MidsummerPlayerScript")
+    {
+    }
+
+    void OnUpdateZone(Player* player, uint32 newZone, uint32 /*newArea*/) override
+    {
+        if (!IsHolidayActive(HOLIDAY_FIRE_FESTIVAL))
+            return;
+
+        auto itr = BonfireStateStore.find(std::make_tuple(player->GetMapId(), newZone, player->GetTeamId()));
+        if ((itr != BonfireStateStore.end()) && (itr->second))
+        {
+            if (!(*(itr->second)))
+            {
+                if (!player->HasAura(SPELL_BONFIRES_BLESSING))
+                    player->CastSpell(player, SPELL_BONFIRES_BLESSING, true);
+
+                return;
+            }
+        }
+
+        player->RemoveAurasDueToSpell(SPELL_BONFIRES_BLESSING);
+    }
+};
+
+struct npc_midsummer_bonfire : public ScriptedAI
+{
+    npc_midsummer_bonfire(Creature* creature) : ScriptedAI(creature)
+    {
+        _isStampedOut  = nullptr;
+        _teamId     = TEAM_NEUTRAL;
+        _type       = BONFIRE_TYPE_NONE;
+        _spellFocus = nullptr;
```

Comment:
only private members should prefix with underscore

Reply:
https://www.azerothcore.org/wiki/cpp-code-standards#naming-of-variables-and-functions

Reply:
My bad, I am just tired.